projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,367 @@
+
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post         = $post_to_check;
+		$default_auth = current_user_can( 'read' );
+
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post = $old_post;
+
+		return $default_auth;
+	}
+
+	/**
+	 * Determines if the current user can access restricted content and builds the restriction_details array.
+	 *
+	 * @param string $guid the video guid.
+	 * @param int    $embedded_post_id the post id.
+	 * @param int    $selected_plan_id the selected plan id if applicable.
+	 *
+	 * @return array
+	 */
+	private function build_restriction_details( $guid, $embedded_post_id, $selected_plan_id ) {
+		$post_to_check = get_post( $embedded_post_id );
+
+		if ( empty( $post_to_check ) ) {
+			$restriction_details = $this->default_video_restriction_details( false );
+			return $this->filter_video_restriction_details( $restriction_details, $guid, $embedded_post_id, $selected_plan_id );
+		}
+
+		$default_auth        = $this->get_default_user_capability_for_post( $post_to_check );
+		$restriction_details = $this->default_video_restriction_details( $default_auth );
+
+		if ( $this->jetpack_memberships_available() ) {
+			$memberships_can_view_post         = \Jetpack_Memberships::user_can_view_post( $embedded_post_id );
+			$restriction_details               = $this->get_subscriber_only_restriction_details( $default_auth );
+			$restriction_details['can_access'] = $memberships_can_view_post;
+			return $this->check_block_level_access(
+				$restriction_details,
+				$guid,
+				$embedded_post_id,
+				$selected_plan_id
+			);
+		}
+
+		return $this->check_block_level_access(
+			$restriction_details,
+			$guid,
+			$embedded_post_id,
+			$selected_plan_id
+		);
+	}
+
+	/**
+	 * Determines if the current user can access restricted block content and updates the restriction_details array.
+	 *
+	 * @param array  $restriction_details the restriction details array.
+	 * @param string $guid the video guid.
+	 * @param int    $embedded_post_id the post id.
+	 * @param int    $selected_plan_id the selected plan id if applicable.
+	 *
+	 * @return array
+	 */
+	private function check_block_level_access( $restriction_details, $guid, $embedded_post_id, $selected_plan_id ) {
+		if ( $this->jetpack_subscriptions_available() && $selected_plan_id > 0 ) {
+			$restriction_details = $this->get_subscriber_only_restriction_details( $restriction_details['can_access'] );
+			$paywall             = \Automattic\Jetpack\Extensions\Premium_Content\subscription_service();
+
+			// Only paid subscribers should be granted access to the premium content.
+			$access_level                      = \Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Token_Subscription_Service::POST_ACCESS_LEVEL_PAID_SUBSCRIBERS;
+			$can_view                          = $paywall->visitor_can_view_content( array( $selected_plan_id ), $access_level );
+			$restriction_details['can_access'] = $can_view;
```

Comment:
Would this be a good place to ensure the video is marked as private, since we know we are dealing with subscriber content here? We could use the `update_video_restriction_details_on_wpcom` function?

Reply:
@roundhill do you mean force-setting the privacy setting to private?

Reply:
Yeah, I was thinking if we make it to this check maybe we could force the video to be set to private so that the JWTs  get requested?

Reply:
after some thought, we are going to address this separately.